models/discordactions.js

```diff
@@ -114,32 +116,56 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
-    const roleIds = groups.map((group) => group.roleid);
 
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
+
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
+
+    const roleIds = groups.map((group) => group.roleid);
     const snapshots = await memberRoleModel.where("roleid", "in", roleIds).get();
-    const roleCount = {};
 
+    const groupsToMemberMaps = [];
+    const roleCount = {};
     snapshots.forEach((doc) => {
-      const roleToMemberMapping = doc.data();
-
-      if (roleCount[roleToMemberMapping.roleid]) {
-        roleCount[roleToMemberMapping.roleid] += 1;
-      } else {
-        roleCount[roleToMemberMapping.roleid] = 1;
+      const groupToMemberMapping = doc.data();
+      if (groupToMemberMapping) {
+        groupsToMemberMaps.push(groupToMemberMapping);
+        roleCount[groupToMemberMapping.roleid] = (roleCount[groupToMemberMapping.roleid] ?? 0) + 1;
```

Comment:
i think we need a comment here for why we are doing this ?

Reply:
+1

Reply:
Added comments.